### DISCUSSION TITLE
Is there a way to create a new data frame from a list

### DISCUSSION THREAD

**Discussion by derrickcchow** [vaexio/vaex#1882] [Q&A] [answered]
Hi,

Is there a equivalent way to create a veax dataframe from a list of values like pandas?

`time_df = pd.DataFrame(row_list,
                               columns=['Data Set', 'Hierarchy One Name', 'H0', 'H1', 'H2', 'H3', 'H4', 'H5',
                                        'Variable Name', 'Variable Name Qualifier', 'Variable Name Sub Qualifier',
                                        'Date of Event', 'Calendar Entry Type', 'Year of Event', 'Quarter',
                                        'Month of Event', 'Week of Event', 'Fiscal Year of Event', 'Fiscal Quarter',
                                        'Fiscal Month of Event', 'Fiscal Week of Event', 'Julian Day',
                                        'Activity Event Id', 'Measure Value', 'Measure Type', 'Partial Period'])`

Derrick

**Comment by JovanVeljanoski** [comment, marked as answer]
Hey!

There is nothing stopping you to create a pandas dataframe first and then import the data in in vaex. 
Something like 

```python
import pandas as pd
import vaex

df_pandas = pd.DataFrame(...) # Create a pandas dataframe in whatever way
df_vaex = vaex.from_pandas(df_pandas)
```

Would that work for you?

**Reply by derrickcchow** [reply]
It does work I am trying to reduce the amount of time I convert the dataframe to Vaex as I do this operation a bunch of times So it slows down the performance.

**Reply by JovanVeljanoski** [reply]
Yeah via pandas will be slower for sure - i thought you were after convenience.

If you care about speed, you will need to transform your data to use something like `vaex.from_arrays(..)` or `vaex.from_items`.

**Reply by derrickcchow** [reply]
okay thank you I will look into doing implementing one of those ways

**Reply by maartenbreddels** [reply]
and `vaex.from_dict`
